alphadia/fdrx/models/two_step_classifier.py

```diff
@@ -83,52 +90,94 @@ def fit_predict(
                 df_predict = df
+
                 self.second_classifier.epochs = 10
 
-            predictions = self._train_and_apply_second_classifier(
+            # train and apply second classifier
+            df_after_second_clf = self._train_and_apply_second_classifier(
                 df_train, df_predict, x_cols, y_col, group_columns
             )
 
-            # Filter results and check for improvement
-            df_filtered = filter_by_qval(predictions, self.second_fdr_cutoff)
-            current_target_count = len(df_filtered[df_filtered["decoy"] == 0])
+            df_filtered = filter_by_qval(df_after_second_clf, self.second_fdr_cutoff)
+            current_target_count = get_target_count(df_filtered)
 
-            if current_target_count < best_precursor_count:
+            if current_target_count < previous_target_count_after_second_clf:
                 logger.info(
-                    f"Stopping training after iteration {i}, "
-                    f"due to decreased target count ({current_target_count} < {best_precursor_count})"
+                    f"Training stopped on iteration {i + 1}. Decrease in target count from "
+                    f"{previous_target_count_after_second_clf:,} to {current_target_count:,}."
                 )
                 return best_result
 
-            best_precursor_count = current_target_count
-            best_result = predictions
+            previous_target_count_after_second_clf = current_target_count
+            best_result = df_after_second_clf
+
+            logger.info(
+                f"{current_target_count:,} precursors found after second classifier, "
+                f"at fdr={self.second_fdr_cutoff}"
+            )
 
-            # Update first classifier if enough confident predictions
+            # update first classifier if enough confident predictions
             if current_target_count > self._min_precursors_for_update:
-                self._update_first_classifier(
-                    df_filtered, df, x_cols, y_col, group_columns
+                logger.info(
+                    f"Sufficient precursors detected ({current_target_count:,} > {self._min_precursors_for_update:,}) "
+                    f"to proceed with update of first classifier"
+                )
+                target_count_after_first_clf, new_classifier = (
+                    self._fit_and_eval_first_classifier(
+                        df_filtered, df, x_cols, y_col, group_columns
+                    )
                 )
+                logger.info(
+                    f"{target_count_after_first_clf:,} precursors found after first classifier, "
+                    f"at fdr={self.first_fdr_cutoff}"
+                )
+
+                if target_count_after_first_clf > previous_target_count_after_first_clf:
+                    logger.info(
+                        f"Updating the first classifier as new target count increased: "
```

Comment:
Maybe "target count" is not very informative to the user who doesn't know about the implementation.
Maybe something like "previous version xxx precursors, current version xxx precursors - updating classifier"